app/frontend/src/features/profile/ProfileTagInput.tsx

```diff
@@ -198,7 +200,19 @@ export default function ProfileTagInput({
             popperDisablePortal: classes.popperDisablePortal,
           }}
           onChange={(_, newValue) => {
-            setPendingValue(newValue);
+            let uniqueValues: Set<string>;
+            if (allowCsv) {
+              const lastIndex = newValue.length - 1;
+              const latestEntry = newValue[lastIndex];
+              const previousEntries = newValue.slice(0, lastIndex);
+              uniqueValues = new Set([
+                ...previousEntries,
+                ...latestEntry.split(",").map((value) => value.trim()),
+              ]);
+            } else {
+              uniqueValues = new Set(newValue);
+            }
+            setPendingValue(Array.from(uniqueValues));
```

Comment:
```suggestion
            setPendingValue(Array.from(
              //remove whitespace-only values
              uniqueValues.filter((value) => !/^\s*$/.test(value))
            ));
```
This is extra but good to have in there